test_keeper_availability_zone/test.py::test_get_availability_zone is flaky

The test is [flaky](https://play.clickhouse.com/play?user=play#CldJVEgKICAgICd0ZXN0X2tlZXBlcl9hdmFpbGFiaWxpdHlfem9uZScgQVMgbmFtZV9zdWJzdHIsCiAgICAxNCBBUyBpbnRlcnZhbF9kYXlzClNFTEVDVAogICAgY2hlY2tfc3RhcnRfdGltZSBBUyBkLAogICAgY291bnQoKSwKICAgIGdyb3VwVW5pcUFycmF5KHB1bGxfcmVxdWVzdF9udW1iZXIpIEFTIHBycywKICAgIGFueShyZXBvcnRfdXJsKQpGUk9NIGNoZWNrcwpXSEVSRSAoKG5vdygpIC0gdG9JbnRlcnZhbERheShpbnRlcnZhbF9kYXlzKSkgPD0gY2hlY2tfc3RhcnRfdGltZSkgQU5EIChwb3NpdGlvbih0ZXN0X25hbWUsIG5hbWVfc3Vic3RyKSA+IDApIEFORCAodGVzdF9zdGF0dXMgSU4gKCdGQUlMJywgJ0VSUk9SJywgJ0ZMQUtZJykpCkdST1VQIEJZIGQKT1JERVIgQlkgZCBERVNDCg==) occasionally:


[Example](https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=80264&sha=2de12d9508d791791e0022cc6b1ee2779b86882c&name_0=PR&name_1=Integration%20tests%20%28aarch64%2C%20distributed%20plan%2C%203%2F4%29)

```
=================================== FAILURES ===================================
__________________________ test_get_availability_zone __________________________
[gw4] linux -- Python 3.10.12 /usr/bin/python3

    def test_get_availability_zone():
        with KeeperClient.from_cluster(cluster, "zoo1") as client1:
            assert client1.get("/keeper/availability_zone") == "az-zoo1"
    
        # Keeper2 set enable_auto_detection_on_cloud to true, but is ignored and <value>az-zoo2</value> is used.
        with KeeperClient.from_cluster(cluster, "zoo2") as client2:
            assert client2.get("/keeper/availability_zone") == "az-zoo2"
            assert "availability_zone" in client2.ls("/keeper")
    
        # keeper3 is not configured with availability_zone value.
        with KeeperClient.from_cluster(cluster, "zoo3") as client3:
            with pytest.raises(Exception):
>               client3.get("/keeper/availability_zone")

test_keeper_availability_zone/test.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
helpers/keeper_utils.py:178: in get
    return self.execute_query(f"get '{path}'", timeout)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <helpers.keeper_utils.KeeperClient object at 0xff436d49c0d0>
query = "get '/keeper/availability_zone'", timeout = 60.0

    def execute_query(self, query: str, timeout: float = 60.0) -> str:
        output = io.BytesIO()
    
        self.proc.stdin.write(query.encode() + b"\n")
        self.proc.stdin.flush()
    
        events = self.poller.poll(timeout)
        if not events:
            raise TimeoutError(f"Keeper client returned no output")
    
        for fd_num, event in events:
            if event & (select.EPOLLIN | select.EPOLLPRI):
                file = self._fd_nums[fd_num]
    
                if file == self.proc.stdout:
                    while True:
                        chunk = file.readline()
                        if chunk.endswith(self.SEPARATOR):
                            break
    
                        output.write(chunk)
    
                elif file == self.proc.stderr:
>                   self.proc.stdout.readline()
E                   Failed: Timeout >900.0s

helpers/keeper_utils.py:159: Failed
---------------------------- Captured stdout setup -----------------------------
Copy common default production configuration from /clickhouse-config. Files: config.xml, users.xml
------------------------------ Captured log setup ------------------------------
```

--- a/tests/integration/helpers/keeper_utils.py
+++ b/tests/integration/helpers/keeper_utils.py
@@ -143,35 +143,37 @@ class KeeperClient(object):
 
     def execute_query(self, query: str, timeout: float = 60.0) -> str:
         output = io.BytesIO()
+        error = b""
 
         self.proc.stdin.write(query.encode() + b"\n")
         self.proc.stdin.flush()
 
-        events = self.poller.poll(timeout)
-        if not events:
-            raise TimeoutError(f"Keeper client returned no output")
+        while True:
+            events = self.poller.poll(timeout)
+            if not events:
+                raise TimeoutError(f"Keeper client returned no output")
 
-        for fd_num, event in events:
-            if event & (select.EPOLLIN | select.EPOLLPRI):
-                file = self._fd_nums[fd_num]
+            for fd_num, event in events:
+                if event & (select.EPOLLIN | select.EPOLLPRI):
+                    file = self._fd_nums[fd_num]
 
-                if file == self.proc.stdout:
-                    while True:
-                        chunk = file.readline()
-                        if chunk.endswith(self.SEPARATOR):
-                            break
+                    if file == self.proc.stdout:
+                        while True:
+                            chunk = file.readline()
+                            if chunk.endswith(self.SEPARATOR):
+                                if error:
+                                    raise KeeperException(
+                                        error.strip().decode()
+                                    )
+                                return output.getvalue().strip().decode()
 
-                        output.write(chunk)
+                            output.write(chunk)
 
-                elif file == self.proc.stderr:
-                    self.proc.stdout.readline()
-                    raise KeeperException(self.proc.stderr.readline().strip().decode())
+                    elif file == self.proc.stderr:
+                        error += self.proc.stderr.readline()
 
-            else:
-                raise ValueError(f"Failed to read from pipe. Flag {event}")
-
-        data = output.getvalue().strip().decode()
-        return data
+                else:
+                    raise ValueError(f"Failed to read from pipe. Flag {event}")
 
     def cd(self, path: str, timeout: float = 60.0):
         self.execute_query(f"cd '{path}'", timeout)

